src/gen/async_generate_data_gemini.py

```diff
@@ -0,0 +1,99 @@
+import asyncio
+import aiohttp
+from typing import List, Dict
+import os
+import sys
+import json
+import numpy as np
+from datasets import load_dataset
+
+
+if os.getenv("COLAB_RELEASE_TAG"):
```

Comment:
I am just curious about the purpose of the Python files. Are they designed to be working as a standalone app? Or are you going to blend these into the existing code base?
- I am asking because some values such as RATE_LIMIT_PER_MINUTE, MODEL_NAME, ... are hard coded.

Reply:
Those can be decided later. 

The purpose of the Python scripts are to generate synthetic data with Gemini out of a notebook environment. The notebooks validate the core logic. The Python scripts should be more or less ready to be in the existing codebase. If you have questions/suggestions around the core logic, feel free to comment. Configuration-related changes aren't a big thing to refactor. Hope we both can agree to that. 

I want to leave the integration work of these scripts with you because you already have that idea as you started it. I necessarily don't agree on a few structures and reinventing the wheel isn't something I want to do. But that said, I still think the standalone scripts can be pretty easily integrated within the current code structure. 

Let me know if you need more clarification.

Reply:
I get your point, and as the list of arguments is growing crazy as we add more steps, I was considering to separate that out into multiple CLI scripts too. 

But still, placing this into the root directory might be a better idea. However, we can do that afterwards!